app/controllers/users_controller.rb

```diff
@@ -1,6 +1,16 @@
 class UsersController < ApplicationController
+
+  before_action :signed_in_user, only: %i[index edit update destroy]
+  before_action :correct_user, only: %i[edit update]
+  before_action :admin_user, only: :destroy
```

Comment:
[Q] Does it mean admin is the only person who can delete the user? But in method `destroy`, `if user == current_user
      flash[:error] = "User cannot be deleted."` , with the case, current_user is admin, so??

Reply:
because, i do not want user delete itself. it's so rick.